### PR TITLE
Build `if-csv`

### PR DESCRIPTION
**Pull request by manushak**
### Types of changes

- [x] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### A description of the changes proposed in the Pull Request

- add `if-csv`  command line tool
- there are 3 cases to test:
    - with output `npm run if-csv -- -m  ./manifests/outputs/plugins/csv-lookup/cloud-metadata/success.yaml  --params cpu-cores-utilized -o ./manifests/outputs/csv-lookup`
    - without output `npm run if-csv -- -m ./manifests/outputs/plugins/csv-lookup/cloud-metadata/success.yaml   --params cpu-cores-utilized`
    - withe pipe `npm run if-run -- -m ./manifests/examples/builtins/csv-lookup/cloud-metadata/success.yml | npm run if-csv --  --params cpu-cores-utilized`

### CODE REVIEW

**Review by jmcook1186 — APPROVED** [review]

File: src/if-csv/index.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import {CsvOptions} from './types/csv';
+
+const IfCsv = async () => {
+  // Call this function with false parameter to prevent log debug messages.
+  debugLogger.overrideConsoleMethods(false);
+
+  const pipedManifest = await parseManifestFromStdin();
+  const {manifest, output, params} = await parseIfCsvArgs();
+  const resolvedManifest = manifest || pipedManifest;
+  let manifestData: Manifest;
```
Comment (narekhovhannisyan) [inline src/if-csv/index.ts:21]:
```
const pipedManifest = await parseManifestFromStdin();
  const {manifest, output, params} = await parseIfCsvArgs();
  const resolvedManifest = manifest || pipedManifest;

  if (resolvedManifest) {
    const manifestData = pipedManifest
      ? ((await YAML.load(pipedManifest!)) as Manifest)
      : await getManifestData(manifest!);
```

File: src/if-csv/util/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    return rawManifest;
+  }
+
+  throw new ManifestValidationError(FAILURE_MESSAGE_OUTPUTS);
+};
+
+/**
+ * Executes a CSV generation based on the provided tree structure, context, output path, and params.
+ */
+export const executeCsv = async (options: CsvOptions) => {
```
Comment (narekhovhannisyan) [inline src/if-csv/util/helpers.ts:31]: execute csv title doesn't describe what this function does. Can be renamed to extract csv by criteria or something like that

File: src/if-csv/util/args.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Checks for `manifest` and `params` flags to be present.
+ */
+export const parseIfCsvArgs = async () => {
+  const {manifest, output, params} = validateAndParseIfCsvArgs();
+
+  if (!params) {
+    throw new ParseCliParamsError(PARAMS_NOT_PRESENT);
+  }
```
Comment (narekhovhannisyan) [inline src/if-csv/util/args.ts:42]: redundant params check

**Review by narekhovhannisyan — CHANGES_REQUESTED** [review]
Please check comments above

File: src/if-csv/util/args.ts (hunk 3 above)
Comment (manushak) [reply src/if-csv/util/args.ts:42]: this `params` is the `if-csv` specific flag

**Review by narekhovhannisyan — APPROVED** [review]

**Pull request merged**